packages/react/src/components/uiComponents.tsx

```diff
@@ -257,7 +257,7 @@ const _UserButton = withClerk(
     const { customPages, customPagesPortals } = useUserProfileCustomPages(props.children, {
       allowForAnyChildren: !!props.__experimental_asProvider,
     });
-    const userProfileProps = Object.assign(props.userProfileProps || {}, { customPages });
+    const userProfileProps = Object.assign({}, props.userProfileProps, { customPages });
```

Comment:
ensures a fresh object is created when merging `customPages` with `userProfileProps`